Added some options

sflis opened the pull request:
Added some extra options for a few scripts to use a different column name for the charge.
Also added the option to choose the fit range for the Gentile fit and at the command line for the extract_spe script.
Don't bother about the first two commits they have nothing to do with the pull request.

watsonjj commented on CHECLabPy/spectrum_fitters/gentile.py:
```diff
@@ -6,7 +6,7 @@
 
 class GentileFitter(SpectrumFitter):
-    def __init__(self, n_illuminations, config_path=None):
+    def __init__(self, n_illuminations, config_path=None, fit_range=None):
```
The range for the fit can be edited by passing a config yaml file containing:

```
range:
- -0.5
- 4
```

Is this okay to use instead of these changes?

sflis replied:
Yeah that is no problem! I did not read through the code thoroughly enough to see that you can set the range through the config file.

watsonjj commented on CHECLabPy/spectrum_fitters/gentile.py:
```diff
@@ -15,13 +15,18 @@ def __init__(self, n_illuminations, config_path=None):
+            self.range = fit_range
 
-        self.add_parameter("norm", None, 0, 100000, fix=True, multi=True)
+        self.add_parameter("norm", None, 0, 1e5, fix=False, multi=True)
```
Could you restore this line?
Parameters can also be changed via the config file. To achieve these changes you did here I would include the following in the config file:
```
limits:
  norm0:
  - 0
  - 1e5
fix:
  norm0: false
```

sflis replied:
I'll do forgot to put them back before committing.

sflis commented on CHECLabPy/core/spectrum_fitter.py:
```diff
@@ -199,11 +199,11 @@ def load_config(self, path):
             for c in self.coeff_names:
                 if 'initial' in d:
                     ini = c
-                    self.initial[ini] = d['initial'].pop(c, self.initial[ini])
```
By using a yaml config file I stumbled upon the fact that yaml does not support scientific notation. The simple and easy solution is convert the values of limits and initial values to floats as we expect to formatted as valid floats in the config file.

sflis commented on CHECLabPy/core/spectrum_fitter.py:
```diff
@@ -199,11 +199,11 @@ def load_config(self, path):
                 if 'initial' in d:
                     ini = c
-                    self.initial[ini] = d['initial'].pop(c, self.initial[ini])
+                    self.initial[ini] = float(d['initial'].pop(c, self.initial[ini]))
```
By using a yaml config file I stumbled upon the fact that yaml does not support scientific notation. The simple and easy solution is convert the values of limits and initial values to floats as we expect to formatted as valid floats in the config file.

sflis commented on CHECLabPy/core/spectrum_fitter.py:
```diff
@@ -200,10 +200,15 @@ def load_config(self, path):
+                    if(isinstance(list_,list)):
+                        self.limits[lim] = tuple([float(l) for l in list_])
+                    else:
+                        self.limits[lim] = list_
```
I realized that my first attempt did not work if the parameter did not have a default value and it was not set in the config file. This less elegant solution checks whether the result from the pop is valid for conversion.

watsonjj commented:
@sflis Could you resolve the conflicts so this branch can be merged?

```
git checkout master
git pull
git checkout <branch>
git merge master
[ ... resolve any conflicts ... ]
git add [files that were conflicted]
git commit
git push
```

sflis commented:
I Have fixed the conflict now and I also added a small tweak so that the norm in the gentile fitter is approximated also by summing events outside the fitting range where the 0th peak is.

watsonjj commented:
I'm not sure about that addition. I don't see how the sum of the charges relates to the normalisation of the fit to the histogram. It should only relate to the number of charges in the histogram.
